Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -854,7 +854,39 @@ public final class PackageBuilder {
         }
 
         // Create and return the right kind of target depending on what kind of sources we found.
-        if sources.hasSwiftSources {
+        if sources.hasSwiftSources && sources.hasClangSources {
+            // First determine the type of module map that will be appropriate for the target based on its header layout.
+            let moduleMapType: ModuleMapType
+
+            if fileSystem.exists(publicHeadersPath) {
+                let moduleMapGenerator = ModuleMapGenerator(targetName: potentialModule.name, moduleName: potentialModule.name.spm_mangledToC99ExtendedIdentifier(), publicHeadersDir: publicHeadersPath, fileSystem: fileSystem)
+                moduleMapType = moduleMapGenerator.determineModuleMapType(observabilityScope: self.observabilityScope)
+            } else if targetType == .library, manifest.toolsVersion >= .v5_5 {
```

Comment:
why the version check?

Reply:
This was copied and pasted from the control flow that returned a `ClangTarget` so I can't say for sure. Initializing a `ClangTarget` requires a `ModuleMapType`. Since a `MixedTarget` wraps a `ClangTarget`, I needed to use this code for the control flow that returned a `MixedTarget`. 

In [558143d](https://github.com/ncooke3/swift-package-manager/pull/2/commits/558143dbe4ceb5e874cc9e0ffedb1f7d5c098cdf), I consolidated this logic into a private method to consolidate its usage.